### PR TITLE
#10 Feat/signup form validate

### PR DESCRIPTION
**Pull request by minhng271**
### Signup form validate
- Check empty
- Check validation
- Show error message

### CODE REVIEW

**Comment by minhng271** [comment]
Uploading Vite + React - Google Chrome 2022-12-21 13-49-25.mp4…

File: react-basic-practice/src/components/common/Input/index.jsx [hunk 1]
```diff
@@ -1,37 +1,52 @@
...
+    handleInput,
+    messageErr,
   } = props;
 
+  const inputRef = useRef();
+
+  const handleChange = (e) => {
+    const inputValue = { [e.target.name]: inputRef.current.value };
+
+    handleInput(inputValue);
```
Comment (MinhNguyen41092) [inline react-basic-practice/src/components/common/Input/index.jsx]: `handleInput` is too vague. Should specify what event you are handling: `handleInputChange`.

File: react-basic-practice/src/components/common/Input/index.jsx [hunk 2]
```diff
@@ -1,37 +1,52 @@
...
+    name,
     cssClasses,
     value,
-    refer,
-    onChange,
     placeholder,
     pattern,
     defaultValue,
+    handleInput,
+    messageErr,
```
Comment (MinhNguyen41092) [inline react-basic-practice/src/components/common/Input/index.jsx]: change to errorMessage.

File: react-basic-practice/src/constants/regexp.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+const REGEXP = {
+  REGEXP_MAIL: /^[\w-\\.]+@([\w-]+\.)+[\w-]{2,4}$/,
+  REGEXP_PASSWORD: /^(?=.*[A-Za-z])(?=.*\d)[A-Za-z\d]{1,}$/g,
+  REGEXP_USER_NAME: /^(?=[a-zA-Z0-9._]{8,20}$)(?!.*[_.]{2})[^_.].*[^_.]$/g,
+};
+
+const MINIMUM_PASSWORD_LENGTH = 8;
```
Comment (MinhNguyen41092) [inline react-basic-practice/src/constants/regexp.js]: This is not a regex.

File: react-basic-practice/src/features/SignUpForm/index.jsx [hunk 4]
```diff
@@ -1,21 +1,84 @@
...
+import { REGEXP } from '../../constants/regexp';
+
 import Button from '../../components/common/Button';
 import Input from '../../components/common/Input';
 import Logo from '../../components/common/Logo';
 import FormGroup from '../../components/FormGroup';
 import './index.css';
 
 const SignUpForm = () => {
+  const initialErrorMsgs = {
```
Comment (MinhNguyen41092) [inline react-basic-practice/src/features/SignUpForm/index.jsx]: Initial variables should be defined outside of SignUpForm function.

File: react-basic-practice/src/features/SignUpForm/index.jsx [hunk 5]
```diff
@@ -1,21 +1,84 @@
...
+    inputEmail: '',
+    inputUserName: '',
+    inputPassword: '',
+  };
+  const initialInput = {
+    inputEmail: '',
+    inputUserName: '',
+    inputPassword: '',
+  };
   const [isSignUpLoading, setIsSignUpLoading] = useState(false);
```
Comment (MinhNguyen41092) [inline react-basic-practice/src/features/SignUpForm/index.jsx:27]: Loading is a common functionality. Should be defined globally.

File: react-basic-practice/src/features/SignUpForm/index.jsx [hunk 6]
```diff
@@ -1,21 +1,84 @@
...
+    }
+  };
+
+  const handleSignUp = (e) => {
+    try {
+      e.preventDefault();
+
+      setIsSignUpLoading(true);
+
+      checkEmpty();
```
Comment (MinhNguyen41092) [inline react-basic-practice/src/features/SignUpForm/index.jsx]: checkEmpty is also a form of validation. Should call this in validation function.

File: react-basic-practice/src/features/SignUpForm/index.jsx [hunk 7]
```diff
@@ -1,21 +1,84 @@
...
+      } else {
+        setErrorMessage((preMsg) => ({
+          ...preMsg,
+          [key]: '',
+        }));
+      }
+    }
+  };
+
+  const checkValidate = () => {
```
Comment (MinhNguyen41092) [inline react-basic-practice/src/features/SignUpForm/index.jsx]: Change to `validateInputs`.

File: react-basic-practice/src/features/SignUpForm/index.jsx [hunk 8]
```diff
@@ -32,33 +95,35 @@ const SignUpForm = () => {
...
+            messageErr={errorMessage.inputUserName}
           />
 
           <Input
             label="Password:"
             inputType="password"
-            inputId="password"
+            name="inputPassword"
             cssClasses="input-form input-password"
             placeholder="Password at least 6 characters"
```
Comment (MinhNguyen41092) [inline react-basic-practice/src/features/SignUpForm/index.jsx]: Should change to `Enter your password`.

**Pull request merged**